health_monitoring/test_health_monitor.py

```diff
@@ -404,3 +404,66 @@ def test(self):
         s = self.get_server("deproxy")
         s.drop_conn_when_receiving_data = True
         time.sleep(1)
+
+
+class TestHmMalformedResponse(tester.TempestaTest):
+    """
+    Test for issue #2147, malformed HM respose led to crash.
+    Fix drop such message, log error and drop it.
+    """
+
+    tempesta = {
+        "config": """
+                listen 80;
+
+                server_failover_http 404 3 10;
+
+                health_check hm0 {
+                    request         "GET / HTTP/1.0\r\n\r\n";
+                    request_url     "/";
+                    resp_code       200;
+                    resp_crc32  0x31f37e9f;
+                    timeout         1;
+                }
+
+                srv_group main {
+                server ${server_ip}:8080;
+
+                health hm0;
+                }
+        """
+    }
+
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8080",
+            "response": "static",
+            "response_content": "HTTP/1.0 200 OK\r\nCon tent-Length:5\r\n\r\nHello",
+        },
+    ]
+
+    def setUp(self):
+        super().setUp()
+        self.klog = dmesg.DmesgFinder(disable_ratelimit=True)
+        self.assert_msg = "Expected nums of warnings in `journalctl`: {exp}, but got {got}"
+        # Cleanup part
+        self.addCleanup(self.cleanup_klog)
+
+    def cleanup_klog(self):
+        if hasattr(self, "klog"):
+            del self.klog
+
+    def test(self):
+        """
+        Waiting for at list one response.
+        """
+        wait_timeout = 2
+
+        self.start_all_services(client=False)
+
+        time.sleep(wait_timeout)
```

Comment:
I think you should use explicit expectation here. 
```suggestion
        util.wait_until(lambda: len(server.requests) != 1)
```